plugins/carla/pluginlistdialog.cpp

```diff
@@ -0,0 +1,1661 @@
+/*
+ * Carla plugin host, adjusted for OBS
+ * Copyright (C) 2011-2023 Filipe Coelho <falktx@falktx.com>
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ */
+
+#include <CarlaBackendUtils.hpp>
+#include <CarlaString.hpp>
+#include <CarlaUtils.h>
+
+#include <QtCore/QDir>
+#include <QtCore/QFileInfo>
+#include <QtCore/QPointer>
+
+#include "pluginlistdialog.hpp"
+#include "pluginrefreshdialog.hpp"
+
+#include "common.h"
+#include "qtutils.h"
+
+CARLA_BACKEND_USE_NAMESPACE
+
+// ----------------------------------------------------------------------------
+// check if the plugin IO makes sense for OBS
+
+template<class T> static bool isSupportedIO(const T &info)
+{
+	return info.cvIns == 0 && info.cvOuts == 0 &&
+	       info.audioIns <= MAX_AV_PLANES &&
+	       info.audioOuts <= MAX_AV_PLANES;
+}
+
+// ----------------------------------------------------------------------------
+// getenv with a fallback value if unset
+
+static inline const char *getEnvWithFallback(const char *const env,
+					     const char *const fallback)
+{
+	if (const char *const value = std::getenv(env))
+		return value;
+
+	return fallback;
+}
+
+// ----------------------------------------------------------------------------
+// Plugin paths (from env vars first, then default locations)
+
+struct PluginPaths {
+	QUtf8String ladspa;
+	QUtf8String lv2;
+	QUtf8String vst2;
+	QUtf8String vst3;
+	QUtf8String clap;
+	QUtf8String jsfx;
+
+	PluginPaths()
+	{
+		// get common env vars
+		const QString HOME = QDir::toNativeSeparators(QDir::homePath());
+
+#if defined(CARLA_OS_WIN)
+		const char *const envAPPDATA = std::getenv("APPDATA");
+		const char *const envLOCALAPPDATA =
+			getEnvWithFallback("LOCALAPPDATA", envAPPDATA);
+		const char *const envPROGRAMFILES = std::getenv("PROGRAMFILES");
+		const char *const envCOMMONPROGRAMFILES =
+			std::getenv("COMMONPROGRAMFILES");
+
+		// Small integrity tests
+		if (envAPPDATA == nullptr) {
+			qFatal("APPDATA variable not set, cannot continue");
+			abort();
+		}
+
+		if (envPROGRAMFILES == nullptr) {
+			qFatal("PROGRAMFILES variable not set, cannot continue");
+			abort();
+		}
+
+		if (envCOMMONPROGRAMFILES == nullptr) {
+			qFatal("COMMONPROGRAMFILES variable not set, cannot continue");
+			abort();
+		}
+
+		const QUtf8String APPDATA(envAPPDATA);
+		const QUtf8String LOCALAPPDATA(envLOCALAPPDATA);
+		const QUtf8String PROGRAMFILES(envPROGRAMFILES);
+		const QUtf8String COMMONPROGRAMFILES(envCOMMONPROGRAMFILES);
+#elif !defined(CARLA_OS_MAC)
+		const QUtf8String CONFIG_HOME(getEnvWithFallback(
+			"XDG_CONFIG_HOME", (HOME + "/.config").toUtf8()));
+#endif
+
+		// now set paths, listing format path spec if available
```

Comment:
I won't block merging over specific comments, but in general I'd prefer if

* Comments use proper grammar (there are definitely other reviewers who will point those errors out)
* Are meaningful beyond a single line explanation of what a block of code does, i.e. explain not only the _what_, but also the _why_
* Do not just contain a URL, but describe what the URL explains and what missing information the link will provide

Again, I won't block over it in every single case, just stating a preference.